samples/drivers/crypto/src/main.c

```diff
@@ -21,8 +21,10 @@ LOG_MODULE_REGISTER(main);
 #define CRYPTO_DRV_NAME CONFIG_CRYPTO_TINYCRYPT_SHIM_DRV_NAME
 #elif CONFIG_CRYPTO_MBEDTLS_SHIM
 #define CRYPTO_DRV_NAME CONFIG_CRYPTO_MBEDTLS_SHIM_DRV_NAME
-#elif CONFIG_CRYPTO_STM32
+#elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32_cryp) && CONFIG_CRYPTO_STM32
```

Comment:
I'm not sure `CONFIG_CRYPTO_STM32` is still required

Reply:
It is not required any more because timycrypt and mbedtls are checked before. I have removed it now.